Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,332 @@
+                name.c_str(),
+                result.IsSuccess() ? (result.GetValue().ToString().c_str()) : ("[Failed]"));
+            AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+            if (result.IsSuccess())
+            {
+                m_status.emplace(name, AZStd::string::format("created as: %s", result.GetValue().ToString().c_str()));
+            }
+            else
+            {
+                m_status.emplace(name, AZStd::string::format("failed : %s", result.GetError().c_str()));
+            }
+        }
+
+        // set transforms of links
+        for (const auto& [name, link_ptr] : links)
+        {
+            const auto this_entry = created_links.at(name);
+            if (this_entry.IsSuccess())
+            {
+                AZ::Transform tf = Utils::GetWorldTransformURDF(link_ptr);
+                auto* entity = AzToolsFramework::GetEntityById(this_entry.GetValue());
+                if (entity)
+                {
+                    auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+                    if (transformInterface)
+                    {
+                        AZ_TracePrintf(
+                            "CreatePrefabFromURDF",
+                            "Setting transform %s %s to [%f %f %f] [%f %f %f %f]",
+                            name.c_str(),
+                            this_entry.GetValue().ToString().c_str(),
+                            tf.GetTranslation().GetX(),
+                            tf.GetTranslation().GetY(),
+                            tf.GetTranslation().GetZ(),
+                            tf.GetRotation().GetX(),
+                            tf.GetRotation().GetY(),
+                            tf.GetRotation().GetZ(),
+                            tf.GetRotation().GetW());
+                        transformInterface->SetWorldTM(tf);
+                    }
+                    else
+                    {
+                        AZ_TracePrintf(
+                            "CreatePrefabFromURDF", "Setting transform failed: %s does not have transform interface", name.c_str());
+                    }
+                }
+            }
+        }
+
+        // set hierarchy
+        for (const auto& [name, link_ptr] : links)
+        {
+            const auto this_entry = created_links.at(name);
+            if (!this_entry.IsSuccess())
+            {
+                AZ_TracePrintf("CreatePrefabFromURDF", "Link %s creation failed", name.c_str());
+                continue;
+            }
+            auto parent_ptr = link_ptr->getParent();
+            if (!parent_ptr)
+            {
+                AZ_TracePrintf("CreatePrefabFromURDF", "Link %s has no parents", name.c_str());
+                continue;
+            }
+            AZStd::string parent_name(parent_ptr->name.c_str(), parent_ptr->name.size());
+            const auto parent_entry = created_links.find(parent_name);
```

Comment:
The `parent_name` and `parent_entry` variables should be lowerCamelCase.
```suggestion
            AZStd::string parentName(parentPtr->name.c_str(), parentPtr->name.size());
            const auto parentEntry = createdLinks.find(parentName);
```